File: Ruby/SpaceCity.rb
#encoding:utf-8

require_relative 'SpaceCityToUI'
require_relative 'Transformation'
require_relative 'SpaceStation'

module Deepspace

    class SpaceCity < SpaceStation

        def initialize(base,rest)
            self.copy(base)
            @base = base
            @collaborators = rest
        end

        def fire
            f = super
            @collaborators.each do |c|
                f += c.fire
            end

            f 
        end

        def protection
            pro = super
            @collaborators.each do |c|
                pro += c.protection
            end

            pro 
        end

        def setLoot(loot)
            super(loot)
            Transformation::NOTRANSFORM
        end

        def to_s
            s = "* BASE:\n"
            s += @base.to_s 
            s +="\n* COLLABORATORS : \n"
            if !@collaborators.empty?
                @collaborators.each do |c|
                    s += c.to_s + "\n"
                    s += "----------------------------------\n"
                end
            end
            s 
        end
        attr_reader :collaborators
            


    end

end

File: Java/deepspace/src/deepspace/SpaceCity.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

import java.util.ArrayList;

/**
 *
 * @author fjsae
 */
public class SpaceCity extends SpaceStation {
    
    private SpaceStation base;
    private ArrayList<SpaceStation> collaborators;
    
    SpaceCity(SpaceStation base, ArrayList<SpaceStation> rest){
        super(base);
        this.base = base;
        this.collaborators = rest;
    }
    
    public ArrayList<SpaceStation> getCollaborators(){
        return collaborators;
    }
    
    @Override
    public float fire(){
        float f = super.fire();
        
        for(SpaceStation c : collaborators){
            f += c.fire();
        }
        
        return f;
    }
    
    @Override
    public float protection(){
        float f = super.protection();
        
        for(SpaceStation c : collaborators){
            f += c.protection();
        }
        
        return f;
    }
    
    @Override
    public Transformation setLoot(Loot loot){
        super.setLoot(loot);
        return Transformation.NOTRANSFORM;
        
    }
}
